database/sqlite/sqlite_metadata.c

```diff
@@ -0,0 +1,1485 @@
+    wc->check_hosts_after    = now_realtime_sec() + METADATA_HOST_CHECK_FIRST_CHECK;
+
+    int shutdown = 0;
+    int in_transaction = 0;
+    int commands_in_transaction = 0;
+    // This can be used in the event loop for all opcodes (not workers)
+    BUFFER *work_buffer = buffer_create(1024);
+    wc->row_id = 0;
+    completion_mark_complete(&wc->init_complete);
+
+    while (shutdown == 0 || (wc->flags & METADATA_WORKER_BUSY)) {
+        RRDDIM *rd;
+        RRDSET *st;
+        RRDHOST *host;
+        uuid_t  *uuid;
+        int rc;
+        DICTIONARY_ITEM *dict_item;
+        BUFFER *buffer;
+
+        worker_is_idle();
+        uv_run(loop, UV_RUN_DEFAULT);
+
+        /* wait for commands */
+        cmd_batch_size = 0;
+        do {
+            if (unlikely(cmd_batch_size >= METADATA_MAX_BATCH_SIZE))
+                break;
+
+            cmd = metadata_deq_cmd(wc, &next_opcode);
+            opcode = cmd.opcode;
+
+            if (unlikely(opcode == METADATA_DATABASE_NOOP && metadata_flag_check(wc, METADATA_FLAG_SHUTDOWN))) {
+                shutdown = 1;
+                continue;
+            }
+
+            ++cmd_batch_size;
+
+            // If we are not in transaction and this command is the same with the next ; start a transaction
+            if (!in_transaction && opcode < METADATA_SKIP_TRANSACTION && opcode == next_opcode) {
+                if (opcode != METADATA_DATABASE_NOOP) {
+                    in_transaction = 1;
+                    db_execute("BEGIN TRANSACTION;");
+                }
+            }
+
+            if (likely(in_transaction)) {
+                commands_in_transaction++;
+            }
+
+            if (likely(opcode != METADATA_DATABASE_NOOP))
+                    worker_is_busy(opcode);
+
+            switch (opcode) {
+                case METADATA_DATABASE_NOOP:
+                case METADATA_DATABASE_TIMER:
+                    break;
+                case METADATA_ADD_CHART:
+                    dict_item = (DICTIONARY_ITEM * ) cmd.param[0];
+                    st = (RRDSET *) dictionary_acquired_item_value(dict_item);
+
+                    rc = sql_store_chart(
+                        &st->chart_uuid,
+                        &st->rrdhost->host_uuid,
+                        string2str(st->parts.type),
+                        string2str(st->parts.id),
+                        string2str(st->parts.name),
+                        rrdset_family(st),
+                        rrdset_context(st),
+                        rrdset_title(st),
+                        rrdset_units(st),
+                        rrdset_plugin_name(st),
+                        rrdset_module_name(st),
+                        st->priority,
+                        st->update_every,
+                        st->chart_type,
+                        st->rrd_memory_mode,
+                        st->entries);
+
+                    if (unlikely(rc))
```

Comment:
From Coveriy:

```sh
*** CID 358495:  Memory - illegal accesses  (UNINIT)
/database/sqlite/sqlite_metadata.c: 1039 in metadata_event_loop()
1033                             st->update_every,
1034                             st->chart_type,
1035                             st->rrd_memory_mode,
1036                             st->entries);
1037     
1038                         if (unlikely(rc))
>>>     CID 358495:  Memory - illegal accesses  (UNINIT)
>>>     Using uninitialized value "rd".
1039                             error_report("Failed to store chart %s", string2str(rd->id));
1040     
1041                         dictionary_acquired_item_release(st->rrdhost->rrdset_root_index, dict_item);
1042                         break;
1043                     case METADATA_ADD_CHART_LABEL:
1044                         dict_item = (DICTIONARY_ITEM * ) cmd.param[0];

** CID 355793:  Resource leaks  (RESOURCE_LEAK)
```

Reply:
Reports chart instead of dimension -- fixing this